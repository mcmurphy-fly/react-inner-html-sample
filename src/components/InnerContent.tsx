import DOMPurify from 'dompurify';
import React, { FunctionComponent } from 'react';
import IProps from './IProps';

const InnerContent: FunctionComponent<IProps> = ({
  content,
  tagName = 'div',
}) => {
  const Wrapper = tagName;

  return (
    <Wrapper
      dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(content) }}
    ></Wrapper>
  );
};

export default InnerContent;
