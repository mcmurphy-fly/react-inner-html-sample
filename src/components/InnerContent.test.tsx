import React from 'react';
import InnerContent from './InnerContent';
import renderer from 'react-test-renderer';

describe('InnerContent', () => {
  test('renders html content in custom tag', () => {
    const tree = renderer
      .create(
        <InnerContent
          content={`
          <h1>Hello</h1>
          <p>bla bla <b>bla</b></p>
      `}
          tagName="section"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders html content in div by default', () => {
    const tree = renderer
      .create(
        <InnerContent
          content={`
          <h1>Hello</h1>
          <p>bla bla <b>bla</b></p>
      `}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('sanitize XSS', () => {
    const tree = renderer
      .create(
        <InnerContent content="<p>abc<iframe//src=jAva&Tab;script:alert(3)>def</p>" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
