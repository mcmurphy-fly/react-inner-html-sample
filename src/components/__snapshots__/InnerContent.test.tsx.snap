// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InnerContent renders html content in custom tag 1`] = `
<section
  dangerouslySetInnerHTML={
    Object {
      "__html": "
          <h1>Hello</h1>
          <p>bla bla <b>bla</b></p>
      ",
    }
  }
/>
`;

exports[`InnerContent renders html content in div by default 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
          <h1>Hello</h1>
          <p>bla bla <b>bla</b></p>
      ",
    }
  }
/>
`;

exports[`InnerContent sanitize XSS 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>abc</p>",
    }
  }
/>
`;
